core-bundle/src/Filesystem/FilesystemUtil.php

```diff
@@ -44,4 +70,51 @@ public static function rewindStream($resource): void
             rewind($resource);
         }
     }
+
+    /**
+     * @param array<string> $uuids
+     *
+     * @return \Generator<FilesystemItem>
+     */
+    private static function doListContentsFromUuids(VirtualFilesystemInterface $storage, array $uuids): \Generator
+    {
+        $paths = [];
+
+        // Keeps track of items; returns true if an item was encountered the first time
+        $track = static function (FilesystemItem $item) use (&$paths): bool {
+            if ($new = !\in_array($path = $item->getPath(), $paths, true)) {
+                $paths[] = $path;
+            }
+
+            return $new;
+        };
+
+        foreach ($uuids as $uuid) {
+            try {
+                if (null === $uuid || null === ($item = $storage->get(Uuid::fromBinary($uuid)))) {
```

Comment:
What if `$uuid` is not binary?

Reply:
Well, it is always like this in the DB fields :smile:. But we could parse from both, especially if we name the helper `…FromSerialized`.

Reply:
Well I'm just asking because the API you build does not say anything about the format that has to be provided :)

Reply:
I went with your suggested name + also improved parsing to handle various formats in fc7284031d3c9959c9cce4e3559a3427a7c2c30e.